src/accelerate/inference.py

```diff
@@ -59,9 +61,26 @@ def build_pipeline(model, split_points, args, kwargs) -> PipelineStage:
 def pippy_forward(forward, *args, **kwargs):
     state = PartialState()
     output = None
+
     if state.num_processes == 1:
         output = forward(*args, **kwargs)
     elif state.is_local_main_process:
+        found_batch_size = None
+        for arg in args:
+            found_batch_size = ignorant_find_batch_size(arg)
+            if found_batch_size is not None:
+                break
+        for kwarg in kwargs.values():
+            found_batch_size = ignorant_find_batch_size(kwarg)
+            if found_batch_size is not None:
+                break
+        if found_batch_size is None:
+            raise ValueError("Could not find batch size from args or kwargs")
+        else:
+            if (found_batch_size % state.num_processes) != 0:
+                slice_to_cut = slice(0, found_batch_size % state.num_processes)
```

Comment:
Still not sure about this one. If we have a `found_batch_size` of 4 with `state.num_processes` of 3, it will do `slice(0,1)` and we will only concatenate one extra input, so we will get in the end a input of batch_size 5 and this is still not % by 3.